Tests/GRPCHTTP2CoreTests/GRPCStreamStateMachineTests.swift

```diff
@@ -0,0 +1,2045 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import NIOCore
+import NIOHPACK
+import XCTest
+
+@testable import GRPCHTTP2Core
```

Comment:
Out of interest, could we do this without a `@testable` import? I always try to aim for not testing internals or use the `package` modifier since that allows to run tests in release mode

Reply:
I'm using the state enum, as well as the header key names which are all internal in the state machine - that's why the `@testable` is needed. 
As for `package`, you mean using that instead of `internal` for the types that have to be used from tests?

Reply:
Yes exactly. The `package` modifier should be able to replace all `@testable` usage or `@_spi(Testing)` in the long run.